lightning-invoice/src/payment.rs

```diff
@@ -1407,11 +1841,29 @@ mod tests {
 		}
 	}
 
-	impl<S: Score> Router<S> for TestRouter {
-		fn find_route(
-			&self, _payer: &PublicKey, route_params: &RouteParameters, _payment_hash: &PaymentHash,
-			_first_hops: Option<&[&ChannelDetails]>, _scorer: &S
+	impl Router for TestRouter {
+		fn find_route<S: Score>(
+			&self, payer: &PublicKey, route_params: &RouteParameters, _payment_hash: &PaymentHash,
+			_first_hops: Option<&[&ChannelDetails]>, scorer: &S
 		) -> Result<Route, LightningError> {
+			// Simulate calling the Scorer just as you would in find_route
+			let route = Self::route_for_value(route_params.final_value_msat);
+			for path in route.paths {
+				for (idx, hop) in path.iter().enumerate() {
+					let usage = ChannelUsage {
+						amount_msat: hop.fee_msat,
```

Comment:
I believe the realistic value here would not just be the fees but the aggregate `fee_msat` from the last hop backwards to this one. You can walk the path backwards to compute this and reverse the expectations in the tests, as that is typically how the scorer would be called.